### PR TITLE
Soft-deprecation message for using _if()/_at()/_all() with a quosure

### PR DESCRIPTION
**Pull request by krlmlr**
Requires https://github.com/r-lib/lifecycle/issues/30.

``` r
library(dplyr)

mutate_at(mtcars[1:3, ], vars(mpg), quo(mean(.)))
#> Warning: The `...` argument of `funs()` can't contain quosures as of dplyr 1.0.0.
#> Please use a one-sided formula, a function, or a function name
#> This warning is displayed once per session.
#> Call `lifecycle::last_warnings()` to see where this warning was generated.
#>                mpg cyl disp  hp drat    wt  qsec vs am gear carb
#> Mazda RX4     21.6   6  160 110 3.90 2.620 16.46  0  1    4    4
#> Mazda RX4 Wag 21.6   6  160 110 3.90 2.875 17.02  0  1    4    4
#> Datsun 710    21.6   4  108  93 3.85 2.320 18.61  1  1    4    1
```

<sup>Created on 2020-03-05 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

### CODE REVIEW

**Comment by hadley** [comment]
Wasn't it deprecated previously?

**Comment by krlmlr** [comment]
Are you suggesting to use `deprecate_warn()` ? Is it worth the effort to be strict about it?

**Comment by hadley** [comment]
No, I just meant the version number was wrong.

**Comment by krlmlr** [comment]
- Changed version in `funs()` message to 0.8.0
- Updated new message to reflect original caller

``` r
library(dplyr)

mutate_at(mtcars[1:3, ], vars(mpg), quo(mean(.)))
#> Warning: The `...` argument of `mutate_at()` can't contain quosures as of dplyr 1.0.0.
#> Please use a one-sided formula, a function, or a function name
#> This warning is displayed once per session.
#> Call `lifecycle::last_warnings()` to see where this warning was generated.
#>                mpg cyl disp  hp drat    wt  qsec vs am gear carb
#> Mazda RX4     21.6   6  160 110 3.90 2.620 16.46  0  1    4    4
#> Mazda RX4 Wag 21.6   6  160 110 3.90 2.875 17.02  0  1    4    4
#> Datsun 710    21.6   4  108  93 3.85 2.320 18.61  1  1    4    1
```

<sup>Created on 2020-03-05 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

I wonder if we should get the name of the original caller through `caller_env()`, instead of passing it in manually.

File: R/colwise-funs.R [hunk 1]
```diff
@@ -24,10 +24,16 @@ as_fun_list <- function(.funs, .env, ...) {
...
-          "Using quosures if funs() is deprecated",
-          "Please use a one-sided formula, a function, or a function name"
-        ))
+        what <- paste0(
+          "dplyr::", .caller, "(", .caller_arg, " = ",
+          "'can\\'t contain quosures')"
+        )
+
+        lifecycle::deprecate_soft(
+          "1.0.0", what,
```
Comment (hadley) [inline R/colwise-funs.R]: Shouldn't this be 0.8.0 too?

Comment (krlmlr) [reply R/colwise-funs.R]: #4330, in 0.8.3 .

File: DESCRIPTION [hunk 2]
```diff
@@ -32,7 +32,7 @@ Depends:
 Imports: 
     ellipsis,
     glue (>= 1.3.0),
-    lifecycle,
+    lifecycle (>= 0.1.0.9000),
```
Comment (romainfrancois) [inline DESCRIPTION]: Does this clash with https://github.com/tidyverse/dplyr/pull/4946 which needs lifecycle 0.2.0

**Comment by hadley** [comment]
Can you please merge/rebase?

**Comment by krlmlr** [comment]
Done, now reduced to a few tweaks.

**Comment by krlmlr** [comment]
:man_facepalming: Misread "superseded" as "soft-deprecated"...

**Comment by krlmlr** [comment]
Done now.

**Pull request merged**